### PR TITLE
feat(internal): Add separate http wrapper for github platform

### PR DESCRIPTION
**Pull request by zharinov**
Separate PR before `platform/github/index.ts` refactoring, which is ready too.

Ref: #5865

### CODE REVIEW

File: lib/util/http/index.ts [hunk 1]
```diff
@@ -66,6 +67,14 @@ export class Http {
         },
       ],
     };
+    if (global.appMode) {
```
Comment (viceice) [inline lib/util/http/index.ts:70]: I think we should always set the user agent

File: lib/util/http/index.ts [hunk 2]
```diff
@@ -28,13 +28,14 @@ export interface HttpResponse<T = string> {
   headers: any;
 }
 
-export class Http {
+export class Http<GetOptions = HttpOptions, PostOptions = HttpPostOptions> {
```
Comment (viceice) [inline lib/util/http/index.ts:31]: Why do we need two types, github is useing one type for both. 🤔

File: lib/util/http/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/util/http/index.ts:70]: Looks like we do, and we'd overwrite the one here: https://github.com/renovatebot/renovate/blob/master/lib/util/got/renovate-agent.ts

Comment (rarkins) [reply lib/util/http/index.ts:70]: I think this whole section can be removed

**Pull request merged**